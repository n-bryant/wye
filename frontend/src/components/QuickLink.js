import React from "react";
import PropTypes from "prop-types";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import Link from "next/link";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

import styles from "./QuickLink.styles";

/**
 * renders a Button wrapped with a Link that serves as a Quick Link
 */
export const QuickLink = props => {
  const classnames = QuickLink.classnames(props);
  const { label, linkHref, linkAs } = props;

  return (
    <React.Fragment>
      {linkAs ? (
        <Link href={linkHref} as={linkAs}>
          <a className={classnames.root()}>
            <Button className={classnames.element("button")}>
              <Typography variant="body1">{label}</Typography>
            </Button>
          </a>
        </Link>
      ) : (
        <Link href={linkHref}>
          <a className={classnames.root()}>
            <Button className={classnames.element("button")}>
              <Typography variant="body1">{label}</Typography>
            </Button>
          </a>
        </Link>
      )}
    </React.Fragment>
  );
};
QuickLink.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}QuickLink`
);
QuickLink.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string,
    button: PropTypes.string
  }),
  // the label to display on the Button
  label: PropTypes.string.isRequired,
  // the href value for the Link
  linkHref: PropTypes.string.isRequired,
  // the as value for the Link
  linkAs: PropTypes.string
};
QuickLink.defaultProps = {
  classes: {}
};

// apply styles
export const StyledQuickLink = withStyles(styles)(QuickLink);

export default StyledQuickLink;
