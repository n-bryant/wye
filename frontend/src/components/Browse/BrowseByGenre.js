import React from "react";
import PropTypes from "prop-types";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import Box from "@material-ui/core/Box";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";

import QuickLink from "../QuickLink";
import styles from "./BrowseByGenre.styles";

/**
 * renders a Button wrapped with a Link that serves as a Quick Link
 */
export const BrowseByGenre = props => {
  const classnames = BrowseByGenre.classnames(props);
  const { genres } = props;

  return (
    <Box my={8} className={classnames.root()}>
      <Typography variant="h2" className={classnames.element("title")}>
        Browse By Genre
      </Typography>
      <Grid container spacing={1}>
        {genres.map((genre, index) => (
          <Grid key={index} item xs={6}>
            <QuickLink
              label={genre === "Massively Multiplayer" ? "MMO" : genre}
              linkHref="/browse/genres/[name]"
              linkAs={`/browse/genres/${genre}`}
              fixedDimensions={true}
            />
          </Grid>
        ))}
      </Grid>
    </Box>
  );
};
BrowseByGenre.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}BrowseByGenre`
);
BrowseByGenre.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string
  }),
  // list of genre names to generate quick links for
  genres: PropTypes.arrayOf(PropTypes.string)
};
BrowseByGenre.defaultProps = {
  classes: {}
};

// apply styles
export const StyledBrowseByGenre = withStyles(styles)(BrowseByGenre);

export default StyledBrowseByGenre;
