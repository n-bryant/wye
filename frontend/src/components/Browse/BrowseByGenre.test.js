import { BrowseByGenre } from "./BrowseByGenre";
import QuickLink from "../QuickLink";

describe("BrowseByGenre", () => {
  it("should render successfully", () => {
    const props = {
      classes: {
        root: "root",
        title: "title"
      },
      genres: ["foo"]
    };
    expect(toJson(shallow(<BrowseByGenre {...props} />))).toMatchSnapshot();
  });

  it("should change the label to MMO if the genre is Massively Multiplayer", () => {
    const props = {
      classes: {
        root: "root",
        title: "title"
      },
      genres: ["Massively Multiplayer"]
    };
    const wrapper = shallow(<BrowseByGenre {...props} />);
    expect(wrapper.find(QuickLink).prop("label")).toBe("MMO");
  });
});
