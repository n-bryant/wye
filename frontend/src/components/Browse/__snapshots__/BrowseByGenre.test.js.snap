// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseByGenre should render successfully 1`] = `
<Styled(MuiBox)
  className="wye-BrowseByGenre root"
  my={8}
>
  <WithStyles(ForwardRef(Typography))
    className="wye-BrowseByGenre__title title"
    variant="h2"
  >
    Browse By Genre
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={1}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="0"
      xs={6}
    >
      <WithStyles(QuickLink)
        fixedDimensions={true}
        label="foo"
        linkAs="/browse/genres/foo"
        linkHref="/browse/genres/[name]"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Styled(MuiBox)>
`;
